### PR TITLE
Migrate user-guide to mkdocs

### PR DESCRIPTION
**Pull request by mazzystr**
Initial commit to migrate user-guide content to new logical hierarchy

> Fixes: #345
> Fixes: #357
> Fixes: #359
> Fixes: #362

### CODE REVIEW

**Comment by mazzystr** [comment]
@dhiller It looks like I'm going to need help with adjusting how Netlify deploys the website.  I don't see a script on how that happens in this repo.  Thanks.

**Comment by dhiller** [comment]
> @dhiller It looks like I'm going to need help with adjusting how Netlify deploys the website. I don't see a script on how that happens in this repo. Thanks.

@mazzystr getting back to this, I am a bit puzzled myself how that works. I always thought there would be no deployment necessary, you would only connect the repo to netlify somehow and the deployment would then just work. @codificat do you know more details about that?

**Comment by dhiller** [comment]
So from what I can see in the repository settings there is an integration for Netlify installed, where the integration has access to the user-guide and the kubevirt.github.io repos. There is not much else to configure.

So what exactly do you want to change?

**Comment by fabiand** [comment]
Thank you!
Can you please rebase the PR?

**Comment by codificat** [comment]
> I always thought there would be no deployment necessary, you would only connect the repo to netlify somehow and the deployment would then just work. @codificat do you know more details about that?

Just a quick note on that: there were/are two kubevirt teams on netlify and our sites were in the "wrong" one - this was fixed and the maintainers in the right team in netlify now have access.

@mazzystr will follow up with the actual purpose and content of this PR (to my understanding, netlify settings are not the target here)

**Comment by mazzystr** [comment]
Yay!  Got rebase to work and merge conflicts fixed. 

Before we merge let's make sure we get a good Netlify preview.  I have Netlify configured correctly (I think?).  It loos like we're getting hung up on mkdocs's python3 requirement.  Netlify forces the use of Ubuntu Xenial 16.04.  I need to tinker with this image to see if python3 is available.

**Comment by mazzystr** [comment]
YAY!  uri, css and javascript problems are fixed!

Netlify build settings are stored in ~/netlify.conf

Netlify preview looks great!  Link is [here](https://deploy-preview-372--kubevirt-user-guide.netlify.app/)

**Comment by mazzystr** [comment]
I'm reapplying do-not-merge/hold label until issues with mkdocs build and GitHub pages are worked out.

**Comment by mazzystr** [comment]
@dhiller user-guide-presubmit-build is still using old code.  Do I need to resubmit this pull req to get the update per project-infra/pull/866?

**Comment by dhiller** [comment]
> @dhiller user-guide-presubmit-build is still using old code. Do I need to resubmit this pull req to get the update per project-infra/pull/866?

Should be fixed with kubevirt/project-infra#876

**Comment by dhiller** [comment]
Looks like there should be a requirements.txt which the presubmit job could use to install the required extensions, so that we don't have to change the job every time a new mkdocs extension is required.

**Comment by mazzystr** [comment]
@dhiller yea we could have a requirements file.  We still have to fix this problem first...
```
 Error: [Errno 13] Permission denied: '/userguide'
/bin/sh: 2: .: Can't open /userguide/bin/activate
```

**Comment by mazzystr** [comment]
I am pretty happy here for the initial commit and happily removing the `do-not-merge/hold` tag

**Review by dhiller — COMMENTED** [review]
Looks good to me in general!

One more thing I'd really like to see would be that the definition of building/deploying the site would be covered by make targets here while the job definition itself would rather just call the make target. This way you can change the way the site is built/deployed without touching the definition of the prow job.

WDYT?

**Comment by mazzystr** [comment]
+1 to the Makefile idea.  I'll work on that immediately.  I thought of it but wasn't sure if we wanted to take that extra step at this time.

**Comment by mazzystr** [comment]
@dhiller  @codificat  Makefile now contains a simple `make build` to be utilized by Prow.  All references back to the website have also been removed.

**Review by dhiller — DISMISSED** [review]
/lgtm
/hold

@codificat any objections from your side?

**Comment by mazzystr** [comment]
Sorry.  I had one more amendment to fix buggy output from the website Makefile sourcing userguide Makefile.  envvar target was displaying bad behavior.

File: .gitignore
Comment (codificat) [inline .gitignore:2]: Are we keeping the generated content in the same branch or having a `source` branch from where the generated site is pushed into the publicly served branch (currently `master`)?

I believe we aim for the latter, in which case the `site` directory would not exist, right?

File: Makefile
Comment (codificat) [inline Makefile:161]: What happens after this? how does the rendered website make it into the branch that gets served?

File: docs/.pages
Comment (codificat) [inline docs/.pages:2]: When I run the site locally (via `make run`) I don't see the welcome page, it gives a 404

File: docs/appendix/membership.md
Comment (codificat) [inline docs/appendix/membership.md:1]: Is this file meant to be in this PR?

File: docs/operations/.pages
Comment (codificat) [inline docs/operations/.pages:16]: I believe that the basic usage page should be towards the top of the list, probably after installation

File: docs/operations/.pages
Comment (codificat) [inline docs/operations/.pages:17]: I believe there is a missing entry for the **Node Eviction** topic (`node-eviction.md`, although the file is under `installation` not `operations`)

**Review by codificat — CHANGES_REQUESTED** [review]
I spotted a few things.

One general doubt/question I have is the reorg that implies moving everything under `docs/`. If I understand correctly, the reason behind this is to separate rendered vs source files, right?
But if that's the case, then my doubt/question becomes: wasn't the plan to separate source vs rendered in two different branches, like in the website?

File: .gitignore
Comment (mazzystr) [reply .gitignore:2]: The `site` directory must preexist.  `mkdocs build` does not create it.  For this piece I'm following what we do with the website where `_site` exists in feature branches but doesn't contain data or if it does it is ignored via `.gitignore`

File: Makefile
Comment (mazzystr) [reply Makefile:161]: @dhiller  Correct me if I'm wrong please.  The tide job handles the merge.

File: docs/appendix/membership.md
Comment (mazzystr) [reply docs/appendix/membership.md:1]: Arg.  Not it is not.  I'll remove that.

File: docs/.pages
Comment (mazzystr) [reply docs/.pages:2]: This is an item that needs to be discussed.  I actually wanted to handle it in a separate issue but now is a good time as any.

Imo, the current README.md needs to be renamed to welcome.md.  That renders the page when browsing directly to `/` and `..../welcome/`.  The README.md at the top level of the repo needs to be a real README containing who/what/when/where/how to operate and develop the site.  If your're good with this I can make that update.

File: docs/operations/.pages
Comment (mazzystr) [reply docs/operations/.pages:17]: Fixed!

File: docs/operations/.pages
Comment (mazzystr) [reply docs/operations/.pages:16]: Fixed.

File: docs/appendix/membership.md
Comment (mazzystr) [reply docs/appendix/membership.md:1]: Fixed.

File: Makefile
Comment (dhiller) [reply Makefile:161]: Yes, the tide job will handle merging the PR into the target branch. But IIUC the docs of mkdocs state that [there's a separate step required](https://metager.de/meta/meta.ger3?eingabe=mkdocs+gh-deploy) to actually deploy your site to GH Pages. 

I always understood netlify only being responsible for showing the previews of PRs to merge, if it also handles display of the merged files then we should be good. If it doesn't deploy the rendered site to gh pages, we need to add another make target that does execute the deployment and adjust the periodic job so that it calls the deploy make target instead.

Makes sense?

File: netlify.toml
Comment (codificat) [inline netlify.toml:3]: shouldn't this look like `make build` to invoke the same command that will be used to render the site?

File: update_changelog.sh
Comment (codificat) [inline update_changelog.sh:22]: As this includes all versions right now (something that might be worth rethinking), probably don't say _Latest_
```suggestion
  echo "# Release notes"
```

File: netlify.conf
Comment (codificat) [inline netlify.conf:8]: Isn't this redundant with `netlify.toml`?

File: mkdocs.yml
Comment (codificat) [inline mkdocs.yml:29]: Where is this displayed?

And, shouldn't it be under `assets/` ?

File: docs/latest_release_notes.md
Comment (codificat) [inline docs/latest_release_notes.md:1]: Isn't this file automatically generated by the update_changelog script?

The script should probably be invoked during build, and the generated page not be part of the repo... right?

**Review by codificat — CHANGES_REQUESTED** [review]
Some minor changes needed I believe.

Also, as I can't comment on binary files, I believe there are some files here that should not be: `docs/grid.png` and `docs/img/favicon.ico` (there are icons in `assets/` but they don't seem to be used? )

File: README.md
Comment (codificat) [inline README.md:3]: This build status from travis is for the website repo, not related to the user-guide...

Comment (mazzystr) [reply README.md:3]: Thanks for that catch.  I forgot to update that url.

File: mkdocs.yml
Comment (mazzystr) [reply mkdocs.yml]: I think I was tinkering here and trying to get the home icon to change.  It's supposed to be favicon and change the icon displayed on the browser tab.

We can update the path.  See b6a9d568ca0

File: docs/latest_release_notes.md
Comment (mazzystr) [reply docs/latest_release_notes.md:1]: update_changelog script does a `head -1` then adds to this markdown.

File: netlify.conf
Comment (mazzystr) [reply netlify.conf]: it is.  I will remove.

File: update_changelog.sh
Comment (mazzystr) [reply update_changelog.sh]: Done!

File: netlify.toml
Comment (mazzystr) [reply netlify.toml:3]: favicon is the icon the browsers display on the tab.

`docs/img/favicon.ico` is the default location for the mkdocs engine if it's not overridden by theme->icon->logo.  I guess I was tinkering here and didn't remove my mess.

Comment (mazzystr) [reply netlify.toml:3]: The rest of the icons in `assets/` already exist in `master`.  They were transferred over.  I don't know if someone stuck icons into content.  I didn't check.

Comment (mazzystr) [reply netlify.toml:3]: Notice python in Netlify lives in /opt/buildhome/python3.7?  That's a non standard path.  I'm not sure python 3.7 is even in $PATH in the Netlify image.  I have to investigate if we can merge this build command into `make build`.  I'm sure we can I just need time.  In the meantime this works.

File: README.md
Comment (codificat) [inline README.md:29]: This seems to be a leftover from an unresolved conflict that should be cleaned up

File: update_changelog.sh
Comment (codificat) [inline update_changelog.sh:28]: As this contains all the releases, let's not call it "latest"... and now that I took a closer look, I'm not sure if "release notes" is a better name than "changelog": the generated content looks more like a changelog than full blown release notes, no?

File: docs/virtual_machines/disks_and_volumes.md
Comment (codificat) [inline docs/virtual_machines/disks_and_volumes.md:406]: hmm... the volume name seems to be `volume1`, no?

File: docs/operations/vmctl.md
Comment (codificat) [inline docs/operations/vmctl.md:78]: Not a native english speaker here, but isn't this supposed to be written: `Kubernetes'` (i.e. not an s before and after the tilde)

**Review by codificat — CHANGES_REQUESTED** [review]
At least one thing to fix in the README. Also 3 more comments/questions/suggestions to review.

File: update_changelog.sh
Comment (mazzystr) [reply update_changelog.sh:28]: I'm good with changing it back to changelog and moving on.

Comment (mazzystr) [reply update_changelog.sh:28]: I withdraw that last comment.  The old docsify layout says Latest Release Notes. I say we keep it and argue the minutia later.

This note did cause me to notice that I forgot to add the page to the table of contents so thank you! :)

File: docs/operations/vmctl.md
Comment (mazzystr) [reply docs/operations/vmctl.md:78]: https://english.stackexchange.com/questions/35654/joness-or-jones

Another example is `I went to the Jones's house`

There is no absolute Oxford English way.  Kubernetes is the project name and singular so the root must be Kubernetes.  That part is easy.  Kubernetes has possession so I add a `'s` and move on.  The plural possessive rule used to be `Jones'`.  I've had many English teachers over the years bounce back and forth.  Everyone is tired of it.  `'s` and continue with life.

File: docs/virtual_machines/disks_and_volumes.md
Comment (mazzystr) [reply docs/virtual_machines/disks_and_volumes.md:406]: not sure what I was thinking here... i unfixed it.

**Comment by codificat** [comment]
ok, it's time to
/lgtm
/approve
that one so the mkdocs set up gets in!

Some follow-up will be in order, including guidance to the other open PRs on how to rebase to these changes

**Comment by dhiller** [comment]
Hrm, the postsubmit failed due to git permission error: https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/logs/push-user-guide-master-build-and-push-to-gh-pages/1362113887071113217

I'll fix this and rerun the job.

**Comment by mazzystr** [comment]
@dhilller Thank you so much!

**Pull request merged**
